Assets/Scripts/StandardSamples/UI/MatchMaking/UISessionsMatchmakingMenu.cs

```diff
@@ -403,5 +434,112 @@ public void OnSessionRefresh(Session session, SessionDetails details)
 
             uiEntry.SetUIElementsFromSessionAndDetails(session, details, this);
         }
+
+        #region UIFriendInteractionSource Implementations
+
+        /* 
+         * REGION NOTES:
+         * 
+         * The UIFriendsMenu popout uses the base class UIFriendInteractionSource to facilitate interactions.
+         * 
+         * NOTE: This current implementation is not complete.
+         * There is functionality suggested by the existance of this window that need to be added.
+         * 
+         * - Users should be able to Join Sessions their friends are in, if that friend is in a Session, and that Session has invites enabled.
+         * - Users that are in a Session who want to interact with a friend that is also in a Session should have UX for choosing Invite or Join
+         * - Users in multiple Sessions should be able to choose which Session to invite their friend to. Currently it uses the most recently joined Session only.
+         * - Users who have friends in multiple Sessions should be able to choose which Session to join
+         * 
+         */
+
+        public override string GetFriendInteractButtonText()
+        {
+            return "Invite";
+        }
+
+        public override void OnFriendInteractButtonClicked(FriendData friendData)
+        {
+            // Get the local Presence Session to invite to
+            if (!GetEOSSessionsManager.TryGetPresenceSession(out Session foundSession) || foundSession.ActiveSession == null)
+            {
+                // Didn't find a presence session, so nothing to invite to
+                Debug.LogError($"{nameof(UISessionsMatchmakingMenu)} ({nameof(OnFriendInteractButtonClicked)}): A friend was chosen to invite to a Session, but no local Presence-enabled Session detected.");
+                return;
+            }
+
+            GetEOSSessionsManager.InviteToSession(foundSession.Name, friendData.UserProductUserId);
+        }
+
+        public override FriendInteractionState GetFriendInteractionState(FriendData friendData)
+        {
+            // First determine if the user is both a friend and online
+            if (!friendData.IsFriend() || !friendData.IsOnline())
+            {
+                return FriendInteractionState.Hidden;
+            }
+
+            if (ownInvitationState == OwnSessionInvitationAbilityState.NoSessionToInviteTo)
+            {
+                return FriendInteractionState.Hidden;
+            }
+
+            if (ownInvitationState == OwnSessionInvitationAbilityState.InvalidSessionToInviteTo)
+            {
+                return FriendInteractionState.Disabled;
+            }
+
+            // The only thing remaining is yes, this user can be interacted with
+            return FriendInteractionState.Enabled;
+        }
+
+        public override void ProcessInformationBeforeFriendsRefresh()
```

Comment:
I have a philosophical problem with this function name.

Functions should be named in a descriptive manner such that they clearly articulate the actions that are taken. Nothing more, and nothing less. 

They should _not_ be named in such a way that makes implications regarding _when_ the function should be called. Such articulations should be made through comments and perhaps supplemented with other documentation.

Reply:
Hmmmm... Going to have a Slack conversation with you on this one!